### DISCUSSION TITLE
Разделение конфигурации на разные группы для tpws

### DISCUSSION THREAD

**Discussion by I-am-all** [bol-van/zapret#253] [Ideas]
Насколько я вижу, сейчас что при запуске twps все параметры применяют разом (с учетом приоритета). Но может оказаться, что для разных хостов нужно применять разные режимы обхода. У меня оказалось, что единственный клиент на телевизоре самсунг работает по TLS 1.2, поэтому только для него помогает mss. при этом это избыточное решение для 90% остального трафика. 
Хорошо бы чтобы можно было разделить конфигурации для разных хостов.
Например, вот так
`tpws --tls --bind-addr 192.168.0.1 --port 999 --disorder --tlsrec=sni --split-pos=2 --hostlist=/opt/home/zapret/russia-youtube.txt | --mss=88 --hostlist=/opt/home/zapret/samsung.txt`

**Comment by bol-van** [comment]
в вашем случае надо фиксировать ип телика
и запускать 2 инстанса тпвс
правилами таблес 2 редиректа

если используются скрипты запрета, то можно делать кастом скрипт
но тут не на раз два для юзера задача

**Reply by I-am-all** [reply]
это, конечно, первое что пришло в голову.  у этого способа несколько недостатков
1. Плодить несколько одинаковых процессов. Приводит к избыточной нагрузке на процессор и память слабого роутера. У меня сейчас tpws занял большую часть оперативки. Я честно говоря, не уверен, что второй процесс влезет
2. Если на каком то IP окажутся разные клиенты (1.2 и 1.3) то эта схема не будет работать

но как временное решение придется использовать

<details>
  <summary>top</summary>
  
 

```
Mem: 96188K used, 30336K free, 364K shrd, 5404K buff, 55956K cached
CPU:  0.1% usr  0.3% sys  0.0% nic 98.0% idle  0.0% io  0.0% irq  1.3% sirq
Load average: 0.00 0.00 0.00 1/44 2132
  PID  PPID USER     STAT   VSZ %VSZ %CPU COMMAND
 2132  2117 admin    R     1688  1.3  0.4 top
 2116   526 admin    S     1572  1.2  0.2 /usr/sbin/dropbear -4
 2067     1 21474836 S    12360  9.7  0.0 /opt/home/zapret/binaries/mips32r1-lsb/tpws --daemon --bind-addr 192.168.0.1 --port 999 --tlsrec=sni --split-pos=1 --hostlist=/opt/home/zapret/russia-youtube.txt --hostlist=/opt/home/zapret/russi
  519     1 admin    S     3928  3.0  0.0 /usr/sbin/httpd -p 80
  539     1 admin    S     1876  1.4  0.0 /sbin/rstats
    1     0 admin    S     1724  1.3  0.0 /sbin/init
  528     1 admin    S     1712  1.3  0.0 /sbin/watchdog
 2044     1 admin    S     1712  1.3  0.0 detect_wan
  484     1 admin    S     1712  1.3  0.0 /sbin/detect_internet
  482     1 admin    S     1712  1.3  0.0 /sbin/detect_link
 2117  2116 admin    S     1696  1.3  0.0 -sh
  542     1 admin    S     1692  1.3  0.0 /bin/sh
 2019     1 admin    S     1684  1.3  0.0 /sbin/udhcpc -i eth2.2 -s /tmp/udhcpc.script -p /var/run/udhcpc0.pid -t4 -T4 -d -O26 -O33 -O121 -O249
  344     1 admin    S     1680  1.3  0.0 /sbin/syslogd -s512 -b0 -S -D -O /tmp/syslog.log
  521     1 admin    S     1680  1.3  0.0 telnetd
  353     1 admin    S     1676  1.3  0.0 /sbin/klogd
  526     1 admin    S     1548  1.2  0.0 /usr/sbin/dropbear -4
 1966     1 nobody   S     1524  1.2  0.0 /usr/sbin/dnsmasq
  536     1 admin    S     1344  1.0  0.0 /bin/lld2d br0
  530     1 admin    S     1236  0.9  0.0 /usr/bin/miniupnpd
  537     1 admin    S     1236  0.9  0.0 /usr/sbin/networkmap -w
    3     2 admin    SW       0  0.0  0.0 [ksoftirqd/0]
  232     2 admin    SW       0  0.0  0.0 [kworker/0:1]
  466     2 admin    SW       0  0.0  0.0 [RtmpMlmeTask]
  348     2 admin    SW       0  0.0  0.0 [usb-storage]
  412     2 admin    SW       0  0.0  0.0 [jbd2/sda-8]
   96     2 admin    SW       0  0.0  0.0 [khubd]
    7     2 admin    SW       0  0.0  0.0 [kworker/u:1]
   87     2 admin    SW<      0  0.0  0.0 [kblockd]
   85     2 admin    SW       0  0.0  0.0 [bdi-default]
    6     2 admin    SW<      0  0.0  0.0 [khelper]
  347     2 admin    SW       0  0.0  0.0 [scsi_eh_0]
  331     2 admin    SW       0  0.0  0.0 [kworker/u:2]
  369     2 admin    SW       0  0.0  0.0 [RtmpCmdQTask]
   86     2 admin    SW<      0  0.0  0.0 [crypto]
  413     2 admin    SW<      0  0.0  0.0 [ext4-dio-unwrit]
   83     2 admin    SW       0  0.0  0.0 [sync_supers]
  122     2 admin    SW       0  0.0  0.0 [kswapd0]
  123     2 admin    SW       0  0.0  0.0 [fsnotify_mark]
  231     2 admin    SW<      0  0.0  0.0 [deferwq]
  465     2 admin    SW       0  0.0  0.0 [RtmpCmdQTask]
    2     0 admin    SW       0  0.0  0.0 [kthreadd]
 2120     2 admin    SW       0  0.0  0.0 [flush-8:0]
    4     2 admin    SW       0  0.0  0.0 [kworker/0:0]


```
</details>

**Reply by bol-van** [reply]
мульти инстанс это нормально в запрете. так и задумано
процессы написаны на c.  это не го, не cpp
без мощных хостлистов каждый процесс ест менее 1 мб памяти
а чтобы не оказались фиксируйте ип адреса по дхцп или статикой

на ближайшие пару месяцев планирую дописать мульти профиль

**Reply by bol-van** [reply]
но на мсс по хосту не расcчитывайте иначе, чем socks rremote dns
реализовать теоретически это возможно, если отложить установку конекта к ресурсу до передачи первого блока данных от клиента
но работает это только на запрос-ответных протоколах
вряд ли буду делать

**Reply by I-am-all** [reply]
> реализовать теоретически это возможно, если отложить установку конекта к ресурсу до передачи первого блока данных от клиента

это некрасиво выглядит технически согласен

**Comment by Karpoffd** [comment]
Есть подвижки по данному вопросу?
У меня тоже разные устройства с разными параметрами работают, например ПК с --oob прекрасно работает, а мобилки с ним нет, вот хочется разделить

**Reply by bol-van** [reply]
здесь надо смотреть трафик и разбираться
хттп и тлс они везде одинаковые
вопрос лишь куда лезут эти устройства
может другие домены, может нестандартные порты
разделять можно средствами  iptables,  если зафиксировать ip
разные стратегии планируются по хостлистам и портам. осенью буду делать
но по  ip источника фильтра не будет
тут рулите иптаблесами и разными инстансами
